File: CPP/Program/Garage.cpp
/*Saya Legi Kuswandi mengerjakan LP4 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang 
telah dispesifikasikan. Aamiin.*/

#pragma once

#include "Car.cpp"
#include "Motor.cpp"
#include "Vehicle.cpp"
#include <string>
#include <vector>
using namespace std;

class Garage {
  private:
    string nama_garasi;
    string luas_garasi;
    vector<Car> mobil;
    vector<Motor> motor;

  public:
    //   Constructor
    Garage() {
    }

    Garage(string nama_garasi, string luas_garasi) {
        this->nama_garasi = nama_garasi;
        this->luas_garasi = luas_garasi;
    }

    Garage(string nama_garasi, string luas_garasi, Motor motor, Car mobil) {
        this->nama_garasi = nama_garasi;
        this->luas_garasi = luas_garasi;
        this->motor.push_back(motor);
        this->mobil.push_back(mobil);
    }

    // Getter and Setter
    void addMobil(Car mobil) {
        this->mobil.push_back(mobil);
    }
    vector<Car> getMobil() {
        return this->mobil;
    }

    void addMotor(Motor motor) {
        this->motor.push_back(motor);
    }

    vector<Motor> getMotor() {
        return this->motor;
    }

    string getNamaGarasi() {
        return nama_garasi;
    }

    void setNamaGarasi(string nama_garasi) {
        this->nama_garasi = nama_garasi;
    }

    string getLuasGarasi() {
        return luas_garasi;
    }

    void setLuasGarasi(string luas_garasi) {
        this->luas_garasi = luas_garasi;
    }

    // Destructor
    ~Garage() {
    }
};

File: Python/Program/Garage.py
from typing import List
from Car import Car
from Motor import Motor

class Garage:
    def __init__(self, nama_garasi="", luas_garasi=""):
        # Inisialisasi atribut garasi
        self.nama_garasi = nama_garasi  # Nama garasi
        self.luas_garasi = luas_garasi  # Luas garasi
        self.mobil = []  # Daftar mobil yang ada di garasi
        self.motor = []  # Daftar motor yang ada di garasi

    # Method untuk menambahkan mobil ke dalam garasi
    def add_mobil(self, mobil):
        self.mobil.append(mobil)

    # Method untuk menambahkan motor ke dalam garasi
    def add_motor(self, motor):
        self.motor.append(motor)

    # Method untuk mendapatkan daftar mobil di garasi
    def get_mobil(self) -> List[Car]:
        return self.mobil

    # Method untuk mendapatkan daftar motor di garasi
    def get_motor(self) -> List[Motor]:
        return self.motor

    # Method untuk mendapatkan nama garasi
    def get_nama_garasi(self) -> str:
        return self.nama_garasi

    # Method untuk mengatur nama garasi
    def set_nama_garasi(self, nama_garasi: str):
        self.nama_garasi = nama_garasi

    # Method untuk mendapatkan luas garasi
    def get_luas_garasi(self) -> str:
        return self.luas_garasi

    # Method untuk mengatur luas garasi
    def set_luas_garasi(self, luas_garasi: str):
        self.luas_garasi = luas_garasi
